src/statsmodels/formula.jl

```diff
@@ -374,12 +374,11 @@ end
     expandcols(trm::Vector)
 Create pairwise products of columns from a vector of matrices
 """
-function expandcols(trm::Vector)
+function expandcols{T<:ModelMatrixContainer}(trm::Vector{T})
     if length(trm) == 1
-        asmatrix(Matrix{Float64}, convert(Array{Float64}, trm[1]))
+        trm[1]
     else
-        a = convert(Array{Float64}, trm[1])
```

Comment:
As far as I can tell, the conversions here are redundant since elements of `trm` are always created by `modelmat_cols`, which already handles this?